lib/stm32/l4/rcc.c

```diff
@@ -458,17 +463,15 @@ uint32_t rcc_get_usart_clk_freq(uint32_t usart)
 {
 	/* Handle values with selectable clocks. */
 	if (usart == LPUART1_BASE) {
-		return rcc_uart_i2c_clksel_freq_hz(rcc_apb2_frequency, RCC_CCIPR_LPUART1SEL_SHIFT);
+		return rcc_uart_i2c_clksel_freq_hz(rcc_apb1_frequency, RCC_CCIPR_LPUART1SEL_SHIFT, RCC_CCIPR);
 	} else if (usart == USART1_BASE) {
-		return rcc_uart_i2c_clksel_freq_hz(rcc_apb1_frequency, RCC_CCIPR_USART1SEL_SHIFT);
+		return rcc_uart_i2c_clksel_freq_hz(rcc_apb2_frequency, RCC_CCIPR_USART1SEL_SHIFT, RCC_CCIPR);
 	} else if (usart == USART2_BASE) {
-		return rcc_uart_i2c_clksel_freq_hz(rcc_apb1_frequency, RCC_CCIPR_USART2SEL_SHIFT);
+		return rcc_uart_i2c_clksel_freq_hz(rcc_apb1_frequency, RCC_CCIPR_USART2SEL_SHIFT, RCC_CCIPR);
 	} else if (usart == USART3_BASE) {
-		return rcc_uart_i2c_clksel_freq_hz(rcc_apb1_frequency, RCC_CCIPR_USART3SEL_SHIFT);
-	} else if (usart == UART4_BASE) {
-		return rcc_uart_i2c_clksel_freq_hz(rcc_apb1_frequency, RCC_CCIPR_UART4SEL_SHIFT);
-	} else {  /* USART5 */
```

Comment:
please don't.  What makes you think uart5 doesn't exist?  See RM0351 for instance?

Reply:
Interesting, RM0394 does not mention a 5th one (that's for STM32L41xxx/42xxx/43xxx/44xxx/45xxx/46xxx). How should this be handled?

Reply:
what do you mean? we support the superset, it's user error to attempt to use a peripheral that doesn't exist.  We have zero interest in having two libraries for different L4s.  (that might be "optimal" but it's unmaintainable in the current state)

Reply:
OK, so implementing the superset and leaving the responsibility to the user is your choice to handle it. That's all I wanted to know... no strong feelings about this - getting the superset correct and complete is hard enough :)

Reply:
Yes, we've been doing superset only since the beginning, even F1 has massive differences in availability.  At least, in 99% of cases, ST makes it a proper superset, not a conflicting set :)